api/include/opentelemetry/global/provider.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "opentelemetry/trace/noop.h"
+#include "opentelemetry/trace/tracer_provider.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+
+using opentelemetry::trace::NoopTracer;
+using opentelemetry::trace::TracerProvider;
+
+namespace opentelemetry
+{
+namespace global
+{
+
+class DefaultTracerProvider final : public TracerProvider
+{
+public:
+  explicit DefaultTracerProvider() : tracer_{new NoopTracer()} {}
+  ~DefaultTracerProvider() { delete tracer_; }
+
+  NoopTracer *const GetTracer(string_view library_name, string_view library_version = "") override
+  {
+    return tracer_;
+  }
+
+private:
+  NoopTracer *const tracer_;
+};
+
+class Provider
+{
+public:
+  static TracerProvider *GetTracerProvider() { return getPtrRef().get(); }
+  static void SetTracerProvider(TracerProvider *tp) { getPtrRef().reset(tp); }
+
+private:
+  static nostd::unique_ptr<TracerProvider> &getPtrRef()
+  {
+    static nostd::unique_ptr<TracerProvider> tracePtr(new DefaultTracerProvider());
```

Comment:
I'm worried that using `unique_ptr` here can lead to issues with order of destructors, e.g. https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2

Should we make this a plain `TracerProvider*` instead?

`sizeof(DefaultTracerProvider)` is 16.

Reply:
Sounds good, pushed up a change.

On a semi-related note, I'm also curious about the ownership structure we want to place on providers passed in - do we assume the global provider will control the lifetime of the provider passed in using `SetTracerProvider`? If so, would it make more sense to also handle the construction of the underlying provider in this global namespace with something like `opentelemetry::global::GetTracerProvider<ConcreteTracerProvider>`? Not totally sure what the idiomatic way to handle this is, though it feels like there is potential to allow users to shoot themselves in the foot by constructing a provider and freeing it after passing it in.

Reply:
In Python, the SDK takes ownernship of `TraceProvider`, the resource will be released as part of the SDK uninitialization - which happens at process exit.